DashAI/back/api/api_v1/endpoints/datasets.py

```diff
@@ -107,7 +155,10 @@ async def upload_dataset(
         os.makedirs(folder_path)
     except FileExistsError as e:
         log.error(e)
-        return {"message": "Dataset name already exists."}
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
```

Comment:
The 500 errors are intended for critical server errors. In this case, it is a request problem, so it would return a 409 Conflict.

Reference: https://stackoverflow.com/questions/3825990/http-response-code-for-post-when-resource-already-exists